test/unit/workDiv/src/WorkDivForKernelTest.cpp

```diff
@@ -0,0 +1,182 @@
+        double var8 = var7 + fmod(var7, 5);
+        double var9 = var8 + fmod(var8, 5);
+        double var10 = var9 + fmod(var9, 5);
+        double var11 = var10 + fmod(var10, 5);
+        double var12 = var11 + fmod(var11, 5);
+        double var13 = var12 + fmod(var12, 5);
+        double var14 = var13 + fmod(var13, 5);
+        double var15 = var14 + fmod(var14, 5);
+        double var16 = var15 + fmod(var15, 5);
+        double var17 = var16 + fmod(var16, 5);
+        double var18 = var17 + fmod(var17, 5);
+        double var19 = var18 + fmod(var18, 5);
+        double var20 = var19 + fmod(var19, 5);
+        double var21 = var20 + fmod(var20, 5);
+        double var22 = var21 + fmod(var21, 5);
+        double var23 = var22 + fmod(var22, 5);
+        double var24 = var23 + fmod(var23, 5);
+        double var25 = var24 + fmod(var24, 5);
+        double var26 = var25 + fmod(var25, 5);
+        double var27 = var26 + fmod(var26, 5);
+        double var28 = var27 + fmod(var27, 5);
+        double var29 = var28 + fmod(var28, 5);
+        double var30 = var29 + fmod(var29, 5);
+        double var31 = var30 + fmod(var30, 5);
+        double var32 = var31 + fmod(var31, 5);
+        double var33 = var32 + fmod(var32, 5);
+        double var34 = var33 + fmod(var33, 5);
+        double var35 = var34 + fmod(var34, 5);
+
+        double sum = var0 + var1 + var2 + var3 + var4 + var5 + var6 + var7 + var8 + var9 + var10 + var11 + var12
+                     + var13 + var14 + var15 + var16 + var17 + var18 + var19 + var20 + var21 + var22 + var23 + var24
+                     + var25 + var26 + var27 + var28 + var29 + var30 + var31 + var32 + var33 + var34 + var35;
+        printf("The sum is %5.2f, the argument is %lu ", sum, sizeOfArray);
+    }
+};
+
+using TestAccs = alpaka::test::EnabledAccs<alpaka::DimInt<1u>, std::uint32_t>;
+
+TEMPLATE_LIST_TEST_CASE("getFunctionAttributes.1D.withIdx", "[workDivKernel]", TestAccs)
+{
+    using Acc = TestType;
+    using Idx = alpaka::Idx<Acc>;
+    using Dim = alpaka::Dim<Acc>;
+    using Vec = alpaka::Vec<Dim, Idx>;
+    using WorkDiv = alpaka::WorkDivMembers<Dim, Idx>;
+    auto const platform = alpaka::Platform<Acc>{};
+    auto const dev = alpaka::getDevByIdx(platform, 0);
+
+    // Kernel allows less than 1024 threads per block if it uses a high number of registers
+    TestKernelWithManyRegisters kernel;
+    size_t const kernelArg = 200ul;
+    auto const& bundeledKernel = alpaka::makeKernelBundle<Acc>(kernel, kernelArg);
+
+    // Test getValidWorkDivForKernel for 1024 threads per grid
+    auto const workDiv = alpaka::getValidWorkDivForKernel(dev, bundeledKernel, Vec{1024}, Vec{1});
+    // Test isValidWorkDivKernel function
+    auto const isWorkDivValidForKernel = alpaka::isValidWorkDivKernel<Acc>(dev, bundeledKernel, workDiv);
+    CHECK(isWorkDivValidForKernel == true);
+
+// If Acc is Cuda or Hip
+#if defined(ALPAKA_ACC_GPU_CUDA_ENABLED) || defined(ALPAKA_ACC_GPU_HIP_ENABLED)
```

Comment:
This is not checking if the Acc is CUDA or HIP.  I do not understand if you like to active the code if CUDA is enabled in CMake or if you like to go into this code only if the ACC is CUDA

Reply:
IMO the macro guards are not required at all.

Reply:
When I remove precompile directives and only use `if constexpr` I get "type undefined" error... Ok I will try to solve by traits someow merging type check and type existence check.

```
if constexpr(std::is_same_v<
                     alpaka::AccGpuUniformCudaHipRt<alpaka::ApiCudaRt, alpaka::Dim<Acc>, alpaka::Idx<Acc>>,
                     Acc>)
```